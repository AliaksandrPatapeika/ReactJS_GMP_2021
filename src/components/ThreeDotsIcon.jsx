import '../less/components/ThreeDotsIcon.less';

import PropTypes from 'prop-types';
import React from 'react';

import threeDotsIconLink from '../img/moreButton.svg';

const ThreeDotsIcon = ({toggleMenu}) => (
  <button type="button" className="threeDotsIcon" onClick={toggleMenu}>
    <img
      src={threeDotsIconLink}
      alt="MovieButton"
    />
  </button>
);

ThreeDotsIcon.propTypes = {
  toggleMenu: PropTypes.func.isRequired
};

export default ThreeDotsIcon;
