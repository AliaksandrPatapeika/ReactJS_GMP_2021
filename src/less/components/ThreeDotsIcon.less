@import "../variables";

.threeDotsIcon {
	opacity: 0;
	background-color: @black;
	position: absolute;
	right: 10px;
	top: 10px;
	border-radius: 50%;
	border: 6px solid @black;
	height: 30px;
	width: 30px;
}
