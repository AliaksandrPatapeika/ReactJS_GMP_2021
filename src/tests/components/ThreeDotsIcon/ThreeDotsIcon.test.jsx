import React from 'react';
import renderer from 'react-test-renderer';

import ThreeDotsIcon from '../../../components/ThreeDotsIcon';

describe('ThreeDotsIcon', () => {
  test('renders correctly', () => {
    const threeDotsIcon = renderer
      .create(<ThreeDotsIcon toggleMenu={Function} />)
      .toJSON();

    expect(threeDotsIcon).toMatchSnapshot();
  });
});
