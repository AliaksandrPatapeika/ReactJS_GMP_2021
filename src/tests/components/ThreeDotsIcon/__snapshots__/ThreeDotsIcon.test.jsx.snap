// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThreeDotsIcon renders correctly 1`] = `
<button
  className="threeDotsIcon"
  onClick={[Function]}
  type="button"
>
  <img
    alt="MovieButton"
    src="test-file-stub"
  />
</button>
`;
